File: leetcode2/CreateTargetArrayintheGivenOrder/a.cpp
/* Copyright (C) 2020 by iamslash */

#include <cstdio>
#include <vector>

using namespace std;

// 4ms 61.03% 8.6MB 30.99%
class Solution {
 public:
  vector<int> createTargetArray(vector<int>& vals, vector<int>& idxs) {
    vector<int> ans;
    int n = vals.size();
    for (int i = 0; i < n; ++i) {
      ans.insert(ans.begin() + idxs[i], vals[i]);
    }
    return ans;
  }
};

File: leetcode2/CreateTargetArrayintheGivenOrder/a.go
/* Copyright (C) 2020 by iamslash */

package main

//             i
// idxs: 0 1 2 2 1
// vals: 0 1 2 3 4
//  ans: 0 1 2 3 0

// 0ms 100.00% 2.2MB 46.43%
// brute force
// O(NlgN) O(N)
func createTargetArray(vals []int, idxs []int) []int {
	n := len(vals)
	ans := make([]int, n)
	for i, idx := range idxs {
		copy(ans[idx+1:i+1], ans[idx:i])
		ans[idx] = vals[i]
	}
	return ans
}
